src/test/kotlin/com/exactpro/th2/check1/rule/RuleFactoryTest.kt

```diff
@@ -413,6 +416,67 @@ class RuleFactoryTest {
         assertEvents()
     }
 
+    @Test
+    fun `setting checkSimpleCollectionsOrder value`() {
+        verifyCheckSimpleCollectionsOrderValue(true, true, true)
+        verifyCheckSimpleCollectionsOrderValue(true, false, true)
+
+        verifyCheckSimpleCollectionsOrderValue(false, true, false)
+        verifyCheckSimpleCollectionsOrderValue(false, false, false)
+
+        verifyCheckSimpleCollectionsOrderValue(null, true, true)
+        verifyCheckSimpleCollectionsOrderValue(null, false, false)
+    }
+
+    private fun verifyCheckSimpleCollectionsOrderValue(
+        requestValue: Boolean?,
+        defaultValue: Boolean,
+        expectedValue: Boolean
+    ) {
+        val streams = createStreams(messages = emptyList())
+        val config = Check1Configuration().apply { setDefaultCheckSimpleCollectionsOrder(defaultValue) }
+        val ruleFactory = RuleFactory(config, streams, clientStub)
+        val rootFilterBuilder = RootMessageFilter.newBuilder().setMessageType("TestMsgType")
+
+        if (requestValue != null) {
+            rootFilterBuilder.setComparisonSettings(
+                RootComparisonSettings.newBuilder().setCheckSimpleCollectionsOrder(BoolValue.newBuilder().setValue(requestValue))
+            )
+        }
+
+        val request = CheckRuleRequest.newBuilder()
+            .setParentEventId(EventID.newBuilder().setId("root").build())
+            .setConnectivityId(ConnectionID.newBuilder().setSessionAlias("test_alias"))
+            .setRootFilter(rootFilterBuilder)
+            .setMessageTimeout(5)
+            .setChainId(ChainID.newBuilder().setId("test_chain_id"))
+            .build()
+
+        val createCheckRule = assertDoesNotThrow {
+            ruleFactory.createCheckRule(request, true)
+        }
+        assertNotNull(createCheckRule) { "Rule cannot be null" }
+
+        assertEquals(
+            expectedValue,
+            createCheckRule.messageFilter.comparatorSettings.isCheckSimpleCollectionsOrder,
+            "Wrong isCheckSimpleCollectionsOrder value in filter: requestValue = $requestValue, defaultValue = $defaultValue"
+        )
+    }
+
+    private val CheckRuleTask.messageFilter: SailfishFilter
+        get() {
+            val messageFilterField = this.javaClass.getDeclaredField("messageFilter")
+            messageFilterField.isAccessible = true
+            return messageFilterField.get(this) as SailfishFilter
+        }
+
+    private fun Check1Configuration.setDefaultCheckSimpleCollectionsOrder(value: Boolean) {
+        val defaultCheckField = this.javaClass.getDeclaredField("defaultCheckSimpleCollectionsOrder")
+        defaultCheckField.isAccessible = true
+        defaultCheckField.setBoolean(this, value)
+    }
```

Comment:
Why do you use accessors to private fields? If you need this then something wrong with the code and we need to change it.

Also, I think we should check the work of the created rule and see if it behaves like we expect

Reply:
Any updates about that?

Reply:
Please, correct tests. We should not use any reflection and access to private fields in tests